packages/flutter/lib/src/material/range_slider.dart

```diff
@@ -0,0 +1,1443 @@
+
+  final CustomSemanticsAction _decreaseStart = const CustomSemanticsAction(label: 'Decrease Min');
+  final CustomSemanticsAction _increaseStart = const CustomSemanticsAction(label: 'Increase Min');
+  final CustomSemanticsAction _decreaseEnd = const CustomSemanticsAction(label: 'Decrease Max');
+  final CustomSemanticsAction _increaseEnd = const CustomSemanticsAction(label: 'Increase Max');
+
+  double get _semanticActionUnit => divisions != null ? 1.0 / divisions : _adjustmentUnit;
+
+  void _increaseStartAction() {
+    if (isEnabled) {
+      onChanged(RangeValues(_increaseValue(values.start), values.end));
+    }
+  }
+
+  void _decreaseStartAction() {
+    if (isEnabled) {
+      onChanged(RangeValues(_decreaseValue(values.start), values.end));
+    }
+  }
+
+  void _increaseEndAction() {
+    if (isEnabled) {
+      onChanged(RangeValues(values.start, _increaseValue(values.end)));
+    }
+  }
+
+  void _decreaseEndAction() {
+    if (isEnabled) {
+      onChanged(RangeValues(values.start, _decreaseValue(values.end)));
+    }
+  }
+
+  double _increaseValue(double value) {
+    return (value + _semanticActionUnit).clamp(0.0, 1.0);
+  }
+
+  double _decreaseValue(double value) {
+    return (value - _semanticActionUnit).clamp(0.0, 1.0);
+  }
+}
+
+/// A callback that formats the numeric values from a [RangeSlider] widget.
+///
+/// See also:
+///
+///  * [RangeSlider.semanticFormatterCallback], which shows an example use case.
+typedef RangeSemanticFormatterCallback = String Function(RangeValues values);
+
+/// Decides which thumbs (if any) should be selected.
+///
+/// The default finds the closest thumb, but if the thumbs are close to each
+/// other, it waits for movement defined by [dx] to determine the selected
+/// thumb.
+///
+/// Override [RangeSlider.thumbSelector] for custom thumb selection.
+typedef RangeThumbSelector = Thumb Function(
+  TextDirection textDirection,
+  RangeValues values,
+  double tapValue,
+  Size thumbSize,
+  Size trackSize,
+  double dx
+);
+
+/// Object for representing range slider thumb values.
+///
+/// This object is passed into [RangeSlider.values] to set its values, and it
+/// is emitted in [RangeSlider.onChange], [RangeSlider.onChangeStart], and
+/// [RangeSlider.onChangeEnd] when the values change.
+class RangeValues {
+  /// Creates pair of start and end values.
+  const RangeValues(this.start, this.end);
+
+  /// The value of the start thumb.
+  ///
+  /// For LTR text direction, the start is the left thumb, and for RTL text
+  /// direction, the start is the right thumb.
+  final double start;
+
+  /// The value of the end thumb.
+  ///
+  /// For LTR text direction, the end is the right thumb, and for RTL text
+  /// direction, the end is the left thumb.
+  final double end;
+
+  @override
+  bool operator ==(Object other) => other is RangeValues && other.start == start && other.end == end;
+
+  @override
+  int get hashCode => hashValues(start, end);
+
+  @override
+  String toString() {
```

Comment:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#override-tostring